### ISSUE TITLE
Public_arm object has no attribute base_offset

### ISSUE THREAD

**Issue by KevinDang74** [tue-robotics/hero_description#3]
```python
[ERROR][/state_machine2][1556050724.968019]: InvalidUserCodeError: Could not execute state 'GET_PLAN' of type '<robot_smach_states.navigation.navigation.getPlan object at 0x7f49c80d4210>': Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/smach/state_machine.py", line 247, in _update_once
    self._remappings[self._current_label]))
  File "/home/kevin/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/navigation.py", line 65, in execute
    constraint = self.constraint_function()
  File "/home/kevin/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/navigate_to_grasp.py", line 38, in generateConstraint
    angle_offset =-math.atan2(arm.base_offset.y(), arm.base_offset.x())
AttributeError: 'PublicArm' object has no attribute 'base_offset'

[ERROR][/state_machine2][1556050724.968702]: State 'NAVIGATE' failed to execute.
[ERROR][/state_machine2][1556050724.969349]: State 'NAVIGATE_TO_GRAB' failed to execute.
[ERROR][/state_machine2][1556050724.970018]: State 'GRAB_ITEM' failed to execute.
[ERROR][/state_machine2][1556050724.970475]: State 'GRAB_TRASH' failed to execute.
[ERROR][/state_machine2][1556050724.970893]: State 'TAKE_OUT' failed to execute.
[ERROR][/state_machine2][1556050724.971467]: Robot exited with <class 'smach.exceptions.InvalidUserCodeError'>,Could not execute state 'GET_PLAN' of type '<robot_smach_states.navigation.navigation.getPlan object at 0x7f49c80d4210>': Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/smach/state_machine.py", line 247, in _update_once
    self._remappings[self._current_label]))
  File "/home/kevin/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/navigation.py", line 65, in execute
    constraint = self.constraint_function()
  File "/home/kevin/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/navigate_to_grasp.py", line 38, in generateConstraint
    angle_offset =-math.atan2(arm.base_offset.y(), arm.base_offset.x())
AttributeError: 'PublicArm' object has no attribute 'base_offset'
,<traceback object at 0x7f49f40ae3b0>
```

AttributeError: 'PublicArm' object has no attribute 'base_offset'

**Comment by LoyVanBeek** [comment]
Can you post a stack-trace where this happened exactly?

**Comment by MatthijsBurgh** [comment]
This is VERY HIGH PRIO

**Comment by alberth** [comment]
Addressed in tue-robotics/tue_robocup#761 as emergency fix.

I think the proper way to fix it is not to expose this kind of information outside the robot. Instead, move these computations into the robot, making it smarter.
